src/main/java/com/avispl/symphony/dal/infrastructure/epos/manager/EposManagerCommunicator.java

```
@@ -0,0 +1,881 @@
+	 */
+	private void checkAuthentication() throws Exception {
+		if (StringUtils.isNullOrEmpty(this.getLogin()) || StringUtils.isNullOrEmpty(this.getPassword())) {
+			throw new FailedLoginException("Username or Password field is empty. Please check device credentials");
+		}
+		if (loginInfo == null || loginInfo.isTokenExpired()) {
+			loginInfo = getLoginInfo();
+		}
+	}
+
+	/**
+	 * Retrieve authentication information by sending POST request to authentication API.
+	 */
+	private LoginInfo getLoginInfo() throws Exception {
+		LoginInfo loginInfo = null;
+		try {
+			// Set up headers
+			HttpHeaders headers = new HttpHeaders();
+			headers.setBasicAuth(this.getLogin(), this.getPassword());
+			headers.setContentType(MediaType.MULTIPART_FORM_DATA);
+
+			// Set up request body
+			MultiValueMap<String, String> body = new LinkedMultiValueMap<>();
+			body.add("grant_type", "client_credentials");
+
+			// Send POST request to device
+			HttpEntity<MultiValueMap<String, String>> request = new HttpEntity<>(body, headers);
+			String loginSubDomain = getCurrentEnvironment().getTokenGenerationSubDomain();
+
+			ResponseEntity<JsonNode> responseEntity = this.obtainRestTemplate()
+					.exchange(String.format(EposManagerUri.REQUEST_TOKEN, createRequestUrl(loginSubDomain, this.defaultHostName)), HttpMethod.POST, request, JsonNode.class);
+			JsonNode response = responseEntity.getBody();
+
+			if (response == null || !response.has(EposManagerConstant.ACCESS_TOKEN)) {
+				throw new ResourceNotReachableException("Failed to retrieve authentication information, endpoint not reachable");
+			}
+
+			String token = response.get(EposManagerConstant.ACCESS_TOKEN).asText();
+			long expiresIn = response.get(EposManagerConstant.EXPIRES_IN).asLong();
+			loginInfo = new LoginInfo(token, expiresIn);
+
+		} catch (HttpClientErrorException e) {
+			if (e.getStatusCode() == HttpStatus.BAD_REQUEST) {
```

please add alternative case here. What if it isn't badrequest specifically - the error will get swallowed